packages/transaction-preview/src/index.ts

```diff
@@ -137,6 +111,95 @@ export const patchProvider = (
   return patchedProvider
 }
 
+const handlePreview = async (
+  transaction: TransactionForSim[],
+  fullProviderRequest?: PatchedEIP1193Provider['request'],
+  req?: {
+    method: string
+    params?: Array<unknown>
+  }
+): Promise<void | unknown> => {
+  try {
+    if (!options) {
+      throw new Error(
+        `Please initialize Transaction Preview package prior to previewing a transaction.
+         You can do this by calling the init function with the appropriate params`
+      )
+    }
+    const preview = await simulateTransactions(options, transaction)
+    if (preview.error.length) {
+      fullProviderRequest(req)
+      handleTPErrors(preview)
+    }
+    if (preview.status !== 'simulated' || !netBalanceChangesExist(preview)) {
+      // If transaction simulation was unsuccessful or balanceChanges do
+      // not exist do not create DOM el
+      return fullProviderRequest(req)
+    }
+    if (app) app.$destroy()
+    app = mountTransactionPreview(preview)
+    options.requireTransactionApproval
+      ? handleRequireApproval(app, fullProviderRequest, req)
+      : fullProviderRequest(req)
+          .then(() => {
+            app.$destroy()
+          })
+          .catch(() => app.$destroy())
+  } catch (e) {
+    fullProviderRequest(req)
+    if (app) app.$destroy()
+    throw new Error(`${e}`)
+  }
+}
+
+export const previewTransaction = async (
+  transaction: TransactionForSim[]
+): Promise<MultiSimOutput> => {
+  try {
+    if (!options) {
+      throw new Error(
+        `Please initialize Transaction Preview package prior to previewing a transaction.
+         You can do this by calling the init function with the appropriate params`
+      )
+    }
+    const preview = await simulateTransactions(options, transaction)
+    if (preview.error.length) {
+      handleTPErrors(preview)
+    }
+    if (preview.status !== 'simulated' || !netBalanceChangesExist(preview)) {
+      // If transaction simulation was unsuccessful or balanceChanges do
+      // not exist do not create DOM el
+      console.error('No net balance changes ocurred from this simulation')
+    }
+    if (app) app.$destroy()
+    app = mountTransactionPreview(preview)
+    return preview
+  } catch (e) {
+    if (app) app.$destroy()
+    throw new Error(`${e}`)
+  }
+}
+
+const handleTPErrors = (preview: MultiSimOutput) => {
+  let internalErrs = preview.internalTransactions.reduce(
+    (acc: string[], tx: InternalTransaction[]) => {
+      if (tx.length) {
+        tx.forEach((t: InternalTransaction) => {
+          if (t.errorReason) acc.push(t.errorReason)
+        })
+      }
+      return acc
+    },
+    []
+  )
```

Comment:
because we like to golf ⛳ 
```suggestion
let internalErrs = preview.internalTransactions.reduce(
(acc: string[], tx: InternalTransaction[]) => 
  tx.length ? [...tx.filter(txn => !!txn.errorReason), ...acc] : acc,
[] 
)
```

Reply:
lol probs would want to test but I saw this simplification and couldn't halp myself

Reply:
Lol its shorter but I would say more readable. I get what its doing but how long would it take someone new to this area to read through and try to figure it out? If its even a few seconds/mins more I always go for readability because of maintainability...dont even need to test it to know what its doing, but I do like the clever stuff 🤓